documents/coredocument.go

```diff
@@ -304,38 +343,22 @@ func (cd *CoreDocument) DocumentRootTree() (tree *proofs.DocumentTree, err error
 		return nil, err
 	}
 
-	// For every signature we create a LeafNode
-	sigProperty := NewLeafProperty(SignaturesField, compactProperties(SignaturesField))
-	sigLeafList := make([]proofs.LeafNode, len(cd.Document.Signatures)+1)
-	sigLengthNode := proofs.LeafNode{
-		Property: sigProperty.LengthProp(proofs.DefaultSaltsLengthSuffix),
-		Salt:     make([]byte, idSize),
-		Value:    []byte(fmt.Sprintf("%d", len(cd.Document.Signatures))),
-	}
-	h := sha256.New()
-	err = sigLengthNode.HashNode(h, true)
+	// Next are leaves from the signature data tree
+	signatureTree, err := cd.getSignatureDataTree()
 	if err != nil {
 		return nil, err
 	}
 
-	sigLeafList[0] = sigLengthNode
-	for i, sig := range cd.Document.Signatures {
-		payload := sha256.Sum256(append(sig.EntityId, append(sig.PublicKey, sig.Signature...)...))
-		leaf := proofs.LeafNode{
-			Hash:     payload[:],
-			Hashed:   true,
-			Property: sigProperty.SliceElemProp(proofs.FieldNumForSliceLength(i)),
-		}
-
-		err = leaf.HashNode(h, true)
-		if err != nil {
-			return nil, err
-		}
-
-		sigLeafList[i+1] = leaf
+	properties := signatureTree.PropertyOrder()
```

Comment:
We do not have a way to get all the leaves from the tree, @vedhavyas is adding that function I will merge after that is done.